client.js

```diff
@@ -0,0 +1,113 @@
+var fuse;
+var nextQuery = null;
+var processing = false;
+var currentQuery = null;
+
+var addItem = function(item) {
+  $("ul.local-storage-list").append("<li><a href=''>" + item + "</a></li>");
+};
+
+$(function() {
+
+  $('#query').focus();
+
+  // populate the left side localStorage list
+  // if the localStorage exists, else hide div
+  if(!localStorage.getItem("searched")) {
+    $("div.local-storage-div").hide();
+  } else {
+    var history = localStorage.getItem("searched");
+    $.each(history.split(","), function(index, item) {
+      addItem(item);
+    });
+  }
+
+  $(document).click(function(e) {
+    console.log("Clicked somewhere in the doc");
+    // figure out if this is a result click
+    if(e.target.classList.contains("result-link")) {
+      // this is a result link
+      var query_value = $("#query").val().trim();
+      if(localStorage.getItem("searched")) {
+        localStorage.setItem("searched", localStorage.getItem("searched") + "," + query_value);
```

Comment:
Only on the next reload of the page.

When the page loads, I get all the items from the LocalStorage and add an appropriate `<ul></ul>` with `<li>` elements. And then I simply append `<li>` elements to this list.